src/input/gamepad/gamepadInfo.js

```diff
@@ -0,0 +1,40 @@
+// @flow
+
+import type {Vec2D} from "../../main/util/Vec2D";
+
+export type StickCardinals = { center : Vec2D, left : number, right : number, up : number, down : number };
+
+type ButtonInfo  = null 
+                 | { kind : "pressed", index : number }
+                 | { kind : "value", index : number, threshold : number }
+                 | { kind : "axis", index : number, threshold : number }
+type StickInfo   = null 
+                 | { kind : "axes", xIndex : number, yIndex : number, cardinals : null | StickCardinals }
+                 | { kind : "value", xIndex : number, yIndex : number, cardinals : null | StickCardinals }
+type TriggerInfo = null 
+                 | { kind : "axis", index : number, min : number, max : number }
+                 | { kind : "value", index : number, min : number, max : number }
+                 | { kind : "light", index : number }
+type DPadInfo    =  null
+                 | { kind : "buttons", upIndex : number, downIndex : number, leftIndex : number, rightIndex : number }
+                 | { kind : "axis", index : number }
```

Comment:
We might need more data for this "1 axis" type of d-pad input, depending on how it works with other controllers that have all the d-pad data shoved into a single axis.